app/src/main/java/com/iic/shopingo/ui/activities/RequestList.java

```diff
@@ -0,0 +1,140 @@
+
+    for (int i = 0; i < 10; ++i) {
+      Request req = new Request();
+      req.photoUrl = "";
+      req.name = "Moshe " + i;
+      req.offerInCents = 100*i;
+      req.location.coords.setLatitude(32.063146);
+      req.location.coords.setLongitude(34.770706);
+      req.location.city = "Tel Aviv";
+      req.location.country = "Israel";
+      req.location.address = "13 Rothschild Ave.";
+      req.location.zipcode = "12345";
+      req.location.phone = "12345";
+      req.items.add("1 Milk");
+      req.items.add("1 Bread");
+      req.items.add("1 Cheese");
+      req.items.add("12 Eggs");
+      requests.add(req);
+    }
+
+
+    list.setOnItemClickListener(this);
+    adapter = new RequestListAdapter(this, requests);
+    list.setAdapter(adapter);
+  }
+
+  @Override
+  public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+    Request req = adapter.getItem(position);
+
+    Intent intent = new Intent(this, RequestDetails.class);
+    intent.putExtra(RequestDetails.EXTRA_REQUEST, req);
+    startActivityForResult(intent, position);
+  }
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+    if (requestCode >= 0) {
+      if (resultCode == RequestDetails.RESULT_ACCEPT || resultCode == RequestDetails.RESULT_REJECT) {
+        Request req = adapter.getItem(requestCode);
+        adapter.removeIndex(requestCode);
+        if (listener != null) {
+          if (resultCode == RequestDetails.RESULT_ACCEPT) {
+            listener.onRequestAccepted(req);
+          } else if (resultCode == RequestDetails.RESULT_REJECT) {
+            listener.onRequestRejected(req);
+          }
+        }
+      }
+    } else {
+      super.onActivityResult(requestCode, resultCode, data);
+    }
+  }
+
+  public static class RequestListAdapter extends BaseAdapter {
+    private List<Request> requests = new ArrayList<>();
+    private LayoutInflater inflater;
+    private Context context;
+
+    public RequestListAdapter(Context context, List<Request> requests) {
+      this.context = context;
+      this.requests = requests;
+    }
+
+    public void removeIndex(int index) {
+      requests.remove(index);
+      notifyDataSetChanged();
+    }
+
+    @Override
+    public int getCount() {
+      return requests.size();
+    }
+
+    @Override
+    public Request getItem(int position) {
+      return requests.get(position);
+    }
+
+    @Override
+    public long getItemId(int position) {
+      return position;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+      RequestListItem item = null;
+      if (convertView != null) {
+        item = (RequestListItem)convertView;
+      } else {
+        item = new RequestListItem(context);
+      }
+      Request req = getItem(position);
+      item.setRequest(null, req.name, req.items.size(), req.offerInCents);
+      return item;
+    }
+  }
+}
```

Comment:
please extract the adapter to a top level class, it's big enough to merit its own file.